### ISSUE TITLE
Add option to set color by third factor level within second factor level

### ISSUE THREAD

**Issue by pat-s** [ndphillips/yarrr#21]
Title says it - a param (may be a list) accepting colors for specific factor levels which are split by types. 

Would be more convenient than specifying all factor levels manually if no color palette is used, especially in the case of many levels.

**Comment by ndphillips** [comment]
Isn't this already implemented? In the plot below, OJ is in blue and VC is in red:

`pirateplot(len ~ supp + dose, data = ToothGrowth)`

![image](https://cloud.githubusercontent.com/assets/8480980/19364078/06fd140e-918d-11e6-9701-70741158b69d.png)

**Comment by pat-s** [comment]
Bad explanation by me. What I mean is coloring by another, third factor variable. The formula would be extended to `pirateplot(len ~ supp + dose + X, data = ToothGrowth)`, where X is only changing colors within `dose`. 

Example: 
* factor 1 = dose
* factor 2 = supp

Assume I have not only `OJ` and `VC` as levels but `OJ1`, `OJ2`, `VC1` and `VC2` within `dose`.
Now I want to color all OJx = "blue" and all VCx = "red".

**Comment by pat-s** [comment]
Here the third factor level would be either "GLMM" or "GAM"

<img width="1215" alt="screenshot at okt 14 09-54-07" src="https://cloud.githubusercontent.com/assets/8430564/19379818/405b3174-91f4-11e6-95e8-3c2be45bd33e.png">

**Comment by ndphillips** [comment]
Ah I see, so this would place the figures differently compared to the existing implementation of a third IV, and to adjust how the colors are distributed?

E.g.;
```
pirateplot(height ~ sex + eyepatch + headband, data = pirates)
```

![image](https://cloud.githubusercontent.com/assets/8480980/19380152/2d6aef8a-91f6-11e6-8539-a2388c2eea19.png)

**Comment by pat-s** [comment]
Ahh I did not know that you already implemented a third IV. Since this is definitely a fine option, may be an option would be a param by which you can set the coloring of the bars based on a factor level. Is this possible? 

So referencing to my example above, the call would be something like

`pirateplot(Value ~ Model + Type, color.by = X, color = c("green", "red"), data = pirates)`

where `X` is the factor column to be colored by and `color` does only apply if `color.by` is specified.

The code for my example above looked like this:

```R
pirateplot(data = smry, formula = Value ~ Model + Type,
           pal = c("green", "green", "red", "red"),
           sortx = "sequential",
           ylim = c(0.5, 1))
```

While this approach is okay if you have just some level (here: 2) it might become problematic for more factor levels of `X`.

So in summary, the option to not split by a third IV but just color by a third IV :)
